sdk/eventhubs/azure_messaging_eventhubs/src/common/management.rs

```diff
@@ -0,0 +1,332 @@
+                    } else {
+                        debug!("Non AMQP error: {}", e);
+                        false
+                    }
+                } else {
+                    debug!("No source error found");
+                    false
+                }
+            }
+            _ => {
+                debug!("Non AMQP error: {}", e);
+                false
+            }
+        }
+    }
+
+    fn should_retry_amqp_error(amqp_error: &AmqpError) -> bool {
+        match amqp_error.kind() {
+            AmqpErrorKind::ManagementStatusCode(code, _) => {
+                debug!("Management operation error: {}", code);
+                match code {
+                    // Retry on 408 (Request Timeout) and 429 (Too Many Requests)
+                    azure_core::http::StatusCode::RequestTimeout
+                    | azure_core::http::StatusCode::TooManyRequests
+                    | azure_core::http::StatusCode::InternalServerError
+                    | azure_core::http::StatusCode::BadGateway
+                    | azure_core::http::StatusCode::ServiceUnavailable
+                    | azure_core::http::StatusCode::GatewayTimeout => true,
+                    _ => false,
+                }
+            }
+            AmqpErrorKind::AmqpDescribedError(described_error) => {
+                debug!("AMQP described error: {:?}", described_error);
+                matches!(
+                    described_error.condition(),
+                    AmqpErrorCondition::ResourceLimitExceeded
+                        | AmqpErrorCondition::ConnectionFramingError
+                        | AmqpErrorCondition::LinkStolen
+                )
+            }
+            _ => {
+                debug!("Other AMQP error: {}", amqp_error);
+                false
+            }
+        }
+    }
+
+    pub async fn get_eventhub_properties(&self, eventhub: &str) -> Result<EventHubProperties> {
+        let response = retry_azure_operation(
+            || async move {
+                let mut application_properties: AmqpOrderedMap<String, AmqpValue> =
+                    AmqpOrderedMap::new();
+                application_properties.insert(EVENTHUB_PROPERTY_NAME.to_string(), eventhub.into());
+                let response = self
+                    .management
+                    .call(EVENTHUB_ENTITY_TYPE.to_string(), application_properties)
+                    .await?;
+                Ok(response)
+            },
+            &self.retry_options,
+            Some(Self::should_retry_management_response),
+        )
+        .await?;
+
+        if !response.contains_key(EVENTHUB_PROPERTY_PARTITION_COUNT) {
+            return Err(EventHubsError::from(ErrorKind::InvalidManagementResponse).into());
+        }
+        let name: String = response
+            .get(EVENTHUB_PROPERTY_NAME)
+            .ok_or_else(|| EventHubsError::from(ErrorKind::InvalidManagementResponse))?
+            .clone()
+            .into();
```

Comment:
`into()` should already effectively clone. Both shouldn't be needed here.

Reply:
The `.clone` turns an `&AmqpValue` into a `AmqpValue`. 

Easily fixed with a `From<&AmqpValue> for String`.